ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -657,16 +649,23 @@ private void executeDdlCommand(DdlCommand ddlCommand) {
     }
   }
 
-  private Pair<AbstractStreamCreateStatement, String> maybeAddFieldsFromSchemaRegistry(
-      AbstractStreamCreateStatement streamCreateStatement,
+  private Pair<Statement, String> maybeAddFieldsFromSchemaRegistry(
+      Statement statement,
+      String statementText,
       Map<String, Object> streamsProperties
   ) {
-    Pair<AbstractStreamCreateStatement, String> avroCheckResult =
-        new AvroUtil().checkAndSetAvroSchema(
-            streamCreateStatement,
-            streamsProperties,
-            ksqlEngine.getSchemaRegistryClient()
-        );
-    return avroCheckResult;
+    if (statement instanceof AbstractStreamCreateStatement) {
+      AbstractStreamCreateStatement streamCreateStatement = (AbstractStreamCreateStatement)
+          statement;
+      Pair<AbstractStreamCreateStatement, String> avroCheckResult =
+          new AvroUtil().checkAndSetAvroSchema(
```

Comment:
Just a thought, but wondering if we should pass `AvroUtil` in to the ctor of `KsqlResource` and add a constructor to `AvroUtil` that takes a `SchemaRegistryClient`? Might make it easier to reason about the dependencies?

Reply:
Can we rename it too? Utility classes should be static only.  Maybe `AvroHelper?`